lib/Dialect/TensorExt/Transforms/CollapseInsertionChains.cpp

```diff
@@ -70,14 +70,15 @@ struct FoldSequentialExtractInsertIntoRotate
       LLVM_DEBUG(llvm::dbgs()
                  << "Insert op " << insertOp
                  << " is not directly operating on an extract op.\n");
-      return failure();
+      return rewriter.notifyMatchFailure(
+          insertOp, "insert op does not source from an extract op");
     }
 
     auto shiftRes = calculateShift(insertOp, extractOp);
     if (failed(shiftRes)) {
       LLVM_DEBUG(llvm::dbgs()
                  << "Insert op " << insertOp << " has incalculable shift\n");
-      return failure();
+      return rewriter.notifyMatchFailure(insertOp, "failed to calculate shift");
```

Comment:
likewise here i think LLVM_DEBUG would be redundant - if someone is already debugging they'd see the match failure